packages/blade/src/components/ActionList/actionListUtils.ts

```diff
@@ -71,6 +83,27 @@ const getActionListProperties = (
       });
       const currentIndex = actionListOptions.length - 1;
 
+      const foundSection = sectionData.find((v) => v.title === currentSection);
+      // push the item in the appropriate bucket
+      if (foundSection) {
+        foundSection?.data.push({
```

Comment:
We're pushing data  into `foundSection` but where are we returning it to? I can see in the else branch, we're pushing data into `sectionData` and returning `sectionData`  to the caller. What we we doing with `foundSection`?

Reply:
foundSection is not the actual data we represent, it's a reference to a single item from `sectionData` which we find and push the data into to build a data structure like this: 

```json
[
  { // <-------- this whole object is what "foundSection" references to, 
                 we find it via the `title` & push more items into the `data` array
    "title": "Recent 1",
    "hideDivider": false,
    "data": [
      {
        "title": "Avocados",
        "value": "Avocados",
        "_index": 4
      },
      {
        "title": "Bananas",
        "value": "Bananas",
        "_index": 5
      },
      {
        "title": "Blueberries",
        "value": "Blueberries",
        "_index": 6
      }
    ]
  },
  {
    "title": "Recent 2",
    "hideDivider": true,
    "data": [
      {
        "title": "Cherries",
        "value": "Cherries",
        "_index": 7
      },
      {
        "title": "Crab apples",
        "value": "Crab apples",
        "_index": 8
      },
      {
        "title": "Jambolan",
        "value": "Jambolan",
        "_index": 9
      }
    ]
  }
]
```

Reply:
I still don't understand where we're using foundSection?

Reply:
foundSection is a temporary variable we use to find and mutate the sectionData. 

Minimal example: 

<img width="650" alt="image" src="https://user-images.githubusercontent.com/35374649/232030774-df738b59-3b87-43d3-996c-7118a6bf1df6.png">